### DISCUSSION TITLE
Satisfying the type checker when accessing `request.sid`

### DISCUSSION THREAD

**Discussion by kalilamodow** [miguelgrinberg/Flask-SocketIO#2085] [Q&A] [answered]
Type checkers give an error message (eg. `Cannot access attribute "sid" for class "Request"`) when accessing `request.sid`. Is there any type-safe workaround, or do I have to define a function like
```python
def getSioSid() -> str:
    sid: str = request.sid  # type: ignore
    return sid # type: ignore
```
?

**Comment by miguelgrinberg** [comment, marked as answer]
Unfortunately the `request.sid` attribute was added long before Flask had typing annotations.

You can trick type checkers with a subclass of the request that declares the `sid` attribute, done inside a `TYPE_CHECKING` block. Something like this:

```python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from flask import Request as _Request

    class Request(_Request):
        sid: str

    request: Request
```
